src/test/java/com/thangle/api/book/BookControllerTest.java

```diff
@@ -34,12 +42,105 @@ void shouldFindAll_OK() throws Exception {
+                .andExpect(jsonPath("$.id").value(book.getId().toString()))
+                .andExpect(jsonPath("$.title").value(book.getTitle()))
+                .andExpect(jsonPath("$.author").value(book.getAuthor()))
+                .andExpect(jsonPath("$.description").value(book.getDescription()))
+                .andExpect(jsonPath("$.createdAt").value(book.getCreatedAt().toString()))
+                .andExpect(jsonPath("$.updatedAt").value(book.getUpdatedAt().toString()))
+                .andExpect(jsonPath("$.image").value(book.getImage()))
+                .andExpect(jsonPath("$.userId").value(book.getUserId().toString()));
+
+        verify(bookService).findById(book.getId());
+    }
+
+    @Test
+    void shouldFindByTitleAuthorDescription_OK() throws Exception {
+        final var book = buildBook();
+
+        when(bookService.find(book.getTitle())).thenReturn(Collections.singletonList(book));
+
+        get(BASE_URL + "/search?searchTerm=" + book.getTitle())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].id").value(book.getId().toString()))
+                .andExpect(jsonPath("$[0].title").value(book.getTitle()))
+                .andExpect(jsonPath("$[0].author").value(book.getAuthor()))
+                .andExpect(jsonPath("$[0].description").value(book.getDescription()))
+                .andExpect(jsonPath("$[0].createdAt").value(book.getCreatedAt().toString()))
+                .andExpect(jsonPath("$[0].updatedAt").value(book.getUpdatedAt().toString()))
+                .andExpect(jsonPath("$[0].image").value(book.getImage()))
+                .andExpect(jsonPath("$[0].userId").value(book.getUserId().toString()));
+    }
+
+    @Test
+    void shouldCreate_OK() throws Exception {
+        final var book = buildBook();
+
+        when(bookService.create(any())).thenReturn(book);
+
+        post(BASE_URL, toBookResponseDTO(book))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(book.getId().toString()))
+                .andExpect(jsonPath("$.title").value(book.getTitle()))
+                .andExpect(jsonPath("$.author").value(book.getAuthor()))
+                .andExpect(jsonPath("$.description").value(book.getDescription()))
+                .andExpect(jsonPath("$.createdAt").value(book.getCreatedAt().toString()))
+                .andExpect(jsonPath("$.updatedAt").value(book.getUpdatedAt().toString()))
+                .andExpect(jsonPath("$.image").value(book.getImage()))
+                .andExpect(jsonPath("$.userId").value(book.getUserId().toString()));
+    }
+
+    @Test
+    void shouldUpdate_OK() throws Exception {
+        final var bookNeedsToBeUpdated = buildBook();
+        final var updatedBook = buildUpdatedBook(bookNeedsToBeUpdated.getId());
+
+        updatedBook.setId(bookNeedsToBeUpdated.getId());
+
+        when(bookService.update(eq(bookNeedsToBeUpdated.getId()), any(Book.class))).thenReturn(updatedBook);
```

Comment:
```suggestion
        when(bookService.update(eq(bookNeedsToBeUpdated.getId()), any())).thenReturn(updatedBook);
```

Tránh dùng any (kể cả any(Something.class))